apps/experiments/models.py

```diff
@@ -639,8 +640,32 @@ def send_bot_message(self, instruction_prompt: str, fail_silently=True):
         bot_message = bot_prompt_for_user(self, prompt_instruction=instruction_prompt)
         try_send_message(self, message=bot_message, fail_silently=fail_silently)
 
+    def get_participant_scheduled_messages(self):
+        from apps.events.models import ScheduledMessage
+
+        messages = ScheduledMessage.objects.filter(participant=self.participant, team=self.team)
```

Comment:
What about adding a link to the experiment to this model?

Reply:
You should also prefetch the action:
```suggestion
        messages = ScheduledMessage.objects.filter(participant=self.participant, team=self.team).select_related("action")
```

Reply:
> What about adding a link to the experiment to this model?

Good idea. For the multi-bot stuff this will be necessary :+1: